src/commons/util/file.rs

```diff
@@ -46,81 +55,95 @@ pub fn file_path(base_path: &Path, file_name: &str) -> PathBuf {
 }
 
 /// Saves a file, creating parent dirs as needed
-pub fn save(content: &[u8], full_path: &Path) -> Result<(), io::Error> {
+pub fn save(content: &[u8], full_path: &Path) -> Result<(), KrillIoError> {
     let mut f = create_file_with_path(full_path)?;
-    f.write_all(content)?;
+    f.write_all(content)
+        .map_err(|e| KrillIoError::new(format!("Could not write to: {}", full_path.to_string_lossy()), e))?;
 
     trace!("Saved file: {}", full_path.to_string_lossy());
     Ok(())
 }
 
 /// Saves an object to json - unwraps any json errors!
-pub fn save_json<O: Serialize>(object: &O, full_path: &Path) -> Result<(), io::Error> {
+pub fn save_json<O: Serialize>(object: &O, full_path: &Path) -> Result<(), KrillIoError> {
     let json = serde_json::to_string(object).unwrap();
     save(&Bytes::from(json), full_path)
 }
 
-/// Loads a files and deserializes as json for the expected type. Maps json
-/// errors to io::Error
-pub fn load_json<O: DeserializeOwned>(full_path: &Path) -> Result<O, io::Error> {
+/// Loads a files and deserializes as json for the expected type. Maps json errors to KrillIoError
+pub fn load_json<O: DeserializeOwned>(full_path: &Path) -> Result<O, KrillIoError> {
     let bytes = read(full_path)?;
-    serde_json::from_slice(&bytes).map_err(|_| io::Error::new(io::ErrorKind::Other, "could not deserialize json"))
+    serde_json::from_slice(&bytes).map_err(|_| {
```

Comment:
Did you intend to ignore the actual error object here rather than include it in the returned error object?

Reply:
good point, adding it